Don't mask invalid URL names in dashboard nav access function.

solarissmoke opened the pull request:
If a `url_name` is invalid, then we should allow the exception to propagate instead of silently discarding that menu item. Currently the exception is silently discarded which makes it very difficult to debug an invalid configuration of `OSCAR_DASHBOARD_NAVIGATION`.

sasha0 commented:
@solarissmoke maybe we could control this via setting, like `OSCAR_PROPAGATE_INVALID_URLNAME_EXCEPTION`? So that if someone has invalid records in the settings, it won't raise exception on existing instance, considering this interception exists for a long time. WDYT?

solarissmoke commented:
Hmm. I was just looking through the forums and this has caused quite a few people headaches because invalid menu configurations are silently ignored instead of raising an exception - it is not easy to debug this. 

So even if we did add a setting I think it might be better to default it to raise exceptions rather than silence them? You're right that it could potentially break some sites however. I am not sure what the best approach is.

sasha0 commented:
I think it's better not to change current behavior and default not to raising an exception for the and version and probably deprecate it. Afterwards, just raise exception without a setting in the future versions.

solarissmoke commented:
@sasha0 I've updated the PR - rather than adding a new setting I've just changed it to log the exception. This way the behaviour is unchanged, but admins will be notified that there is a problem.

sasha0 commented:
Let's maybe document this to release notes with a message that in the future versions exception will be propagated, not intercepted.

solarissmoke commented:
Added a comment to the release notes.
